File: Baekjoon/Python/BOJ_PS/Sorting/Q1431.py
# 시리얼 번호 / 정렬
import sys
input = sys.stdin.readline
n = int(input())
serials = [input().strip() for _ in range(n)]

def get_sum(s):
    sum = 0
    for c in s:
        if c.isdigit():
            sum += int(c)
    return sum

serials.sort(key= lambda x: (len(x), get_sum(x), x))
for serial in serials:
    print(serial)

File: Baekjoon/Swift/BOJ_PS/BOJ_PS/Sorting/Q1431.swift
//
//  Q1431.swift
//  BOJ_PS
//
//  Created by 김태훈 on 2021/06/03.
//  시리얼 번호 / 정렬

var serials = [String]()
for _ in 0 ..< Int(readLine()!)! {
    serials.append(readLine()!)
}

func getSum(_ s: String) -> Int {
    let str = Array(s).map { String($0) }
    var sum = 0
    for char in str {
        if Int(char) != nil {
            sum += Int(char)!
        }
    }
    return sum
}

serials.sort {
    if $0.count == $1.count {
        if getSum($0) == getSum($1) {
            return $0 < $1
        } else {
            return getSum($0) < getSum($1)
        }
    } else {
        return $0.count < $1.count
    }
}

serials.forEach { print($0) }
